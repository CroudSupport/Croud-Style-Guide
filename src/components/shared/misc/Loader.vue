<template>
    <div class="ui inverted dimmer" :class="{ active: loading }">
        <div class="ui text loader" :class="{ [size]: true }">
            <slot>
                Loading
            </slot>
        </div>
    </div>
</template>

<script>
   /**
    * A loading spinner combined with a dimmer
    *
    * @see https://semantic-ui.com/elements/loader.html
    * @example ./croud-loader.md
    */
    export default {
        name: 'croud-loader',

        props: {
           /**
            * Whether the loader is active or not
            */
            loading: {
                type: Boolean,
                required: true,
            },

           /**
            * Size of the loader
            * @see https://semantic-ui.com/elements/loader.html#size
            */
            size: {
                type: String,
                default: 'large',
            },
        },
    }
</script>
