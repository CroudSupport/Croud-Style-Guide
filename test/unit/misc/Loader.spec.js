import Vue from 'vue'
import Loader from '../../../src/components/shared/misc/Loader'

const Constructor = Vue.extend(Loader)

const vm = new Constructor({
    propsData: {
        loading: true,
        size: 'small',
    },
}).$mount()

describe('Loader', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    describe('not loading', () => {
        beforeEach(() => {
            vm.loading = false
        })

        it('should match the snapshot', () => {
            expect(vm.$el).toMatchSnapshot()
        })
    })
})
