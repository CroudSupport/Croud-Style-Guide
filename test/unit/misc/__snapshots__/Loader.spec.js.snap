// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader not loading should match the snapshot 1`] = `
<div
  class="ui inverted dimmer"
>
  <div
    class="ui text loader small"
  >
    Loading
  </div>
</div>
`;

exports[`Loader should match the snapshot 1`] = `
<div
  class="ui inverted dimmer active"
>
  <div
    class="ui text loader small"
  >
    Loading
  </div>
</div>
`;
